softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/FRouterHandler.java

```diff
@@ -0,0 +1,29 @@
+package com.softgroup.common.router.api;
+
+import com.softgroup.common.protocol.Request;
+import com.softgroup.common.protocol.Response;
+import com.softgroup.common.router.api.factory.HandlerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public abstract class FRouterHandler implements Handler {
+
+
+    @Override
+    public String getName() {
+        return "mainRouter";
+    }
+
+
+    @Autowired
+    HandlerFactory<AbstractRouterHandler> routerHandlerHandlerFactory;
```

Comment:
@AlexBort First - we must inject interface, not implementation. Next - this must be in abstract router handler